src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,137 @@
 
-        if(!proprietaireService.verifyPassword(password, proprietaireDTO.getId())){
+        if (proprietaireDTO != null && !proprietaireService.verifyPassword(password, proprietaireDTO.getId())) {
             throw new ProprietaireNotFoundException();
+        } else if (administrateurDTO != null && !administrateurService.verifyPassword(password, administrateurDTO.getId())) {
+            throw new AdministrateurNotFoundException();
         }
 
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if (uuid == null){
+        String uuid = tokenCacheService.getFromCache(id);
+        if (uuid == null) {
             uuid = UUID.randomUUID().toString();
-            tokenCacheService.addToCache(proprietaireDTO.getId() ,uuid);
+            tokenCacheService.addToCache(id, uuid);
         }
 
-        String token = jwtHelper.generateToken(email, uuid);
+        String token = jwtHelper.generateToken(email, uuid, isAdmin);
 
         return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
+                id,
+                token,
+                isAdmin
         );
-
     }
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
+    public AuthLoginResponseDTO auth(@NonNull String token) {
+
+        Entry<Long, Boolean> entry = verifyToken(token, false);
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
         return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
+                entry.getKey(),
+                token,
+                entry.getValue()
         );
 
     }
 
-    public void logoutProprietaire(@NonNull String token){
+    public void logout(@NonNull String token) {
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
+        Entry<Long, Boolean> entry = verifyToken(token, false);
 
-        tokenCacheService.addToCache(proprietaireDTO.getId(), UUID.randomUUID().toString());
+        tokenCacheService.addToCache(entry.getKey(), UUID.randomUUID().toString());
 
     }
 
-    public ProprietaireDTO verifyToken(@NonNull String token){
+    public Entry<Long, Boolean> verifyToken(@NonNull String token, boolean shouldBeAdmin) {
         ProprietaireDTO proprietaireDTO;
-        try{
-            proprietaireDTO = proprietaireService.getProprietaireByEmail(jwtHelper.extractEmail(token));
-        } catch (ProprietaireNotFoundException ex){
+        AdministrateurDTO administrateurDTO;
+        Long id;
+        String email;
+        boolean isAdmin = jwtHelper.extractAdmin(token);
+        if (shouldBeAdmin && !isAdmin) {
             throw new TokenExpireOrInvalidException();
         }
 
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if(uuid == null){
+        if (isAdmin) {
+            try {
+                administrateurDTO = administrateurService.getAdministrateurByEmail(jwtHelper.extractEmail(token));
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 100 characters (found 114).